src/Platforms/OMPTarget/ompBLAS.cpp

```diff
@@ -488,12 +488,89 @@ ompBLAS_status copy_batched(ompBLAS_handle&                   handle,
                             const int                         n,
                             const std::complex<double>* const x[],
                             const int                         incx,
-                            std::complex<double>* const y[],
+                            std::complex<double>* const       y[],
                             const int                         incy,
                             const int                         batch_count)
 {
   return copy_batched_impl(handle, n, x, incx, y, incy, batch_count);
 }
 #endif
+
+template<typename T>
+ompBLAS_status copy_batched_offset_impl(ompBLAS_handle& handle,
```

Comment:
BLAS APIs have offset mechanism built-in. Why is this needed?

Reply:
I mean the caller needs to add the offset to x and y arrays without the need of second API.

Reply:
could you share a link to an implementation that does that?
how would the batched equivalent of [this call](https://github.com/anbenali/qmcpack/blob/53bbc039245701bd0059e2ef01666b5f8e13273f/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.2.cpp#L468) work?

I see how it would work if we are just passing a pointer to the start of the data and if we also pass a stride (assuming it is all contiguous), but if we are passing a list of pointers then I would think we'd need to make a new list where each element is incremented by the offset.

Reply:
yes, it was mostly just for convenience to avoid the need to build the new list